maskdepth_generator.py

```diff
@@ -0,0 +1,368 @@
+import torch
```

Comment:
Is everything here so different that it can't fit in the other generator's code?

Reply:
No actually adding this maskdepth_generator.py was not my intention for this pull request, plus it's not the one from the right branch.

Reply:
ok so you'll add a commit to the PR?

Reply:
I'll try and revert my commit for this file